src/hdmf/container.py

```diff
@@ -1,19 +1,88 @@
+import numpy as np
 from abc import abstractmethod
 from uuid import uuid4
 from six import with_metaclass
-from .utils import docval, getargs, ExtenderMeta
+from .utils import docval, get_docval, call_docval_func, getargs, ExtenderMeta
 from warnings import warn
 
 
-class Container(with_metaclass(ExtenderMeta, object)):
+class AbstractContainer(with_metaclass(ExtenderMeta, object)):
 
     _fieldsname = '__fields__'
 
-    # @docval({'name': 'container_source', 'type': str, 'doc': 'source of this Container', 'default': None},
-    #         {'name': 'object_id', 'type': str, 'doc': 'UUID4 unique identifier for this Container', 'default': None},
-    #         {'name': 'parent', 'type': str, 'doc': 'parent Container for this Container', 'default': None})
+    __fields__ = tuple()
```

Comment:
Here, AbstractContainer has a `__fields__` and a `__fields` prop. I suggest changing `__fields__` to `__props__` like in #100